c++/src/Reader.cc

```diff
@@ -1433,6 +1538,7 @@ namespace orc {
   }
 
   void ReaderImpl::startNextStripe() {
+    reader.reset(); // ColumnReaders use lots of memory; free old memory first
```

Comment:
Sorry, I missed that it was the reset on the unique_ptr rather than the ColumnReader. 

In my profiling, one of the major costs for files with little stripe sizes is rebuilding the ColumnReader and the associated buffer. At some point, we need to clean it up so that it reuses the current ColumnReader and buffers.